### PR TITLE
bluetooth: Fix bt_conn->id handling

### LINKED ISSUE #15698: bluetooth: bt_conn: No proper ID handling
**Describe the bug**
bt_conn->id is probably not handled properly (using hardcoded value - zero) - noticed errors connected with inproper BLE connection handling while using directed advertising (e.g. unable to properly refer to the connection).

**Expected behavior**
ID should be updated correctly for BLE connections (it should not have hardcoded value of zero).

**Impact**
 It may cause errors in applications - e.g. problems with direct advertising.

### PR DESCRIPTION
**Pull request by MarekPieta**
Change fixes conn->id handling in le connections. 

Fixes #15698

### CODE REVIEW

**Comment by jhedberg** [comment]
Looks generally fine, but you've misspelled "Bluetooth" in your commit message.

**Review by jhedberg — CHANGES_REQUESTED** [review]
The change looks fine to me, however (as I mentioned in my earlier commit) you need to fix the typo in the commit message

**Comment by jhedberg** [comment]
Also, please add a `Fixes #15698` for traceability and so that the issue gets auto-closed when this gets merged

**Comment by carlescufi** [comment]
As @jhedberg said. I've added the "Fixes" in the description myself.

**Comment by MarekPieta** [comment]
Fixed typo and added `Fixes #15698` to commit message

**Review by jhedberg — APPROVED** [review]

File: subsys/bluetooth/host/conn.c [hunk 1]
```diff
@@ -2061,6 +2061,7 @@ struct bt_conn *bt_conn_create_slave_le(const bt_addr_le_t *peer,
 		return NULL;
 	}
 
+	conn->id = param->id;
```
Comment (pdunaj) [inline subsys/bluetooth/host/conn.c]: I think this should be set by `bt_conn_add_le(id, peer)` (we should add `id` as an argument to this function). Other places where this function seems to be connected with central role but if we don't extend the API now somebody can create a bug in the future.

I am not the maintainer of this stack so I don't want to impose my opinion here so no red x :)

Comment (jhedberg) [reply subsys/bluetooth/host/conn.c]: @pdunaj agreed, however that seems like an independent change, it's essentially refactoring which affects quite a lot of code, whereas the current patch in this PR is fixing the missing initialisation of conn->id. I wouldn't mind adding the refactoring patch to this PR though.

Comment (pdunaj) [reply subsys/bluetooth/host/conn.c]: Good idea. But in this case if you are fine we can merge this one first to have bug fixed. Then refactor can be done in separate PR.

**Review by pdunaj — APPROVED** [review]

File: subsys/bluetooth/host/conn.c (hunk 1 above)
Comment (jhedberg) [reply subsys/bluetooth/host/conn.c]: @pdunaj sounds like a plan

**Pull request merged**